libc-bottom-half/sources/__main_void.c

```diff
@@ -3,6 +3,7 @@
 #include <sysexits.h>
 
 // The user's `main` function, expecting arguments.
+__attribute__((__weak__))
```

Comment:
Why is this needed in particular for the shared build? ?  Perhaps add a comment?

Reply:
...  I wonder if this file should be logically part of crt1?

Reply:
wasi-libc doesn't use musl's crt1.c; it uses libc-bottom-half/crt/crt1-command.c and libc-bottom-half/crt/crt1-reactor.c. Putting the `__main_argc_argc` symbol there sounds reasonable to me.

Reply:
I've moved both the `__main_void` definition and the `__main_argc_argv` declaration to crt1-command.c.